### PR TITLE
Use an option object instead of positional arguments

### PR DESCRIPTION
**Pull request by mcollina**
Ref: https://github.com/pinojs/pino/issues/785

This is a semver-major / breaking change

### CODE REVIEW

**Comment by mcollina** [comment]
@davidmarkclements  @jsumners PTAL

File: README.md [hunk 1]
```diff
@@ -43,22 +43,20 @@ for (var i = 0; i < 10; i++) {
 
 ## API
 
-### SonicBoom(String|Number, [minLength], [sync])
+### SonicBoom(opts) String|Number, [minLength], [sync])
```
Comment (jsumners) [inline README.md]: This line seems wrong now. How about:

```suggestion
### SonicBoom(opts)
```

File: index.js [hunk 2]
```diff
@@ -45,11 +45,15 @@ function openFile (file, sonic) {
...
-function SonicBoom (fd, minLength, sync) {
+function SonicBoom (opts) {
   if (!(this instanceof SonicBoom)) {
-    return new SonicBoom(fd, minLength, sync)
+    return new SonicBoom(opts)
   }
 
+  var { fd, dest, minLength, sync } = opts || {}
+
+  fd = fd || dest
```
Comment (jsumners) [inline index.js]: This looks like it could be `fd = undefined || undefined`. Is that accurate?

Comment (mcollina) [reply index.js]: Yes, this is validated down at line 70. I prefer to keep this double behavior (as it was before).

**Pull request merged**